lesson_13/lesson_13_ilas_lorena/Inheritance.py

```diff
@@ -0,0 +1,61 @@
+import math
+
+
+class PerimeterMixin:
+
+    def perimeter(self):
+        return f'The perimeter is {sum(self.sides)}'
+
+
+class Polygons:
+
+    def __init__(self, *args):
+        self.sides = args
+
+    def __str__(self):
+        n_of_sides = len(self.sides)
+        return '{} is the number of sides'.format(n_of_sides)
+
+    def display(self):
+        for side_index, length in enumerate(self.sides, start=1):
+            print('Side {} with length: {}'.format(side_index, length))
+
+
+class Triangle(PerimeterMixin, Polygons):
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def area(self):
+        s1, s2, s3 = self.sides
+        s_p = sum(self.sides) / 2
+        return f'The triangle area is {math.sqrt(s_p * (s_p - s1) * (s_p - s2) * (s_p - s3))}'
+
+
+class Square(Polygons):
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def area(self):
+        side, *_ = self.sides
+        return f'The square area is {side ** 2}'
+
+    @staticmethod
+    def from_area(area):
```

Comment:
This method is supposed to return a new Square object. The logic of getting the sides is correct, but with the sides, we need to create a new object. In order to create a new object we need the class object (cls).

What we can use from python to achieve this?

Reply:
@classmethod? :)

Reply:
That is correct :D. Can you please make sure the classmethod is returning a new instance. `cls(sq,sq,sq,sq)`

Reply:
> That is correct :D. Can you please make sure the classmethod is returning a new instance. `cls(sq,sq,sq,sq)`

Sure, done!

Reply:
> > That is correct :D. Can you please make sure the classmethod is returning a new instance. `cls(sq,sq,sq,sq)`
> 
> Sure, done!

It is better to return just `cls(sq,sq,sq,sq)`. If you put it in a string it can't be used as an object

Reply:
Sure, is it ok now?

Reply:
yes, thanks :smile: